### PR TITLE
[BD-26] Initial special exams extension lib with timed exam feature for learning app

### PR DESCRIPTION
**Pull request by TamoshaytisV**
This PR adds initial Sequence Wrapper component with start exam instructions page and functionality.
Sequence Wrapper component has its own state (redux) to keep all exam related data independently

Updated the way to build library as a es module (like in frontend-platform)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @TamoshaytisV! I've created [OSPR-5753](https://openedx.atlassian.net/browse/OSPR-5753) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.




:warning: We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://openedx.org/cla)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

File: docs/decisions/0002-feature-based-application-organization.rst [hunk 1]
```diff
@@ -0,0 +1,213 @@
+2. Feature-based Application Organization
```
Comment (idegtiarov) [inline docs/decisions/0002-feature-based-application-organization.rst:1]: This decision is taken from the template app and not need to be here.

**Review by idegtiarov — CHANGES_REQUESTED** [review]

**Comment by natabene** [comment]
@TamoshaytisV Please add [BD-26] to this PR since it seems to be a part of the Blended project. Please ensure to add this tag to all your future PRs within the project.

File: docs/decisions/0002-feature-based-application-organization.rst (hunk 1 above)
Comment (TamoshaytisV) [reply docs/decisions/0002-feature-based-application-organization.rst:1]: @idegtiarov this is fixed in https://github.com/edx/frontend-lib-special-exams/pull/2 already, once merged
will update it here also

**Comment by TamoshaytisV** [comment]
@Zacharis278 @idegtiarov please continue review

File: src/SpecialExamWrapper/Sequence.jsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import {
+  getExamData,
+  getAttemptData,
+  startExam,
+  store,
+} from './data';
+
+const mapCoursewareStateToProps = (state) => {
+  const { courseware } = state;
+  return { courseId: courseware.courseId };
```
Comment (idegtiarov) [inline src/SpecialExamWrapper/Sequence.jsx:16]: Why do we need additional const { courseware } if we can return { courseId: state.course.Id} or I've missed something?

File: src/SpecialExamWrapper/Sequence.test.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    examDuration: 30,
+    attempt: {},
+    examStarted: false,
+  },
+});
+
+test('SequenceExamWrapper renders successfully', () => {
+  const sequence = {
+    id: 'block-v1:test+test+test+type@sequential+block@5b1bb1aaf6d34e79b213aa37422b4743',
+    isTimeLimited: true,
```
Comment (idegtiarov) [inline src/SpecialExamWrapper/Sequence.test.jsx:40]: Could we also add a test that sequence is rendered in case isTimeLimited: false?

**Review by idegtiarov — CHANGES_REQUESTED** [review]
Some minor nits from me and it seems that our tests do not cover negative cases, positive are also has a lack of the tests. Let's consider how we will increase test coverage.

File: docs/decisions/0001-library-rationale.rst [hunk 4]
```diff
@@ -0,0 +1,19 @@
+2. Ideas behind the Special Exams library
+-----------------------------------------
+
+Special Exams library provides set of components to enable users pass exams in the learning micro-fromtend application
+(frontend-app-learning).
```
[the remark this answers is not in the document]
Comment (TamoshaytisV) [reply docs/decisions/0001-library-rationale.rst]: added

File: docs/decisions/0001-library-rationale.rst [hunk 5]
```diff
@@ -0,0 +1,19 @@
+2. Ideas behind the Special Exams library
+-----------------------------------------
+
+Special Exams library provides set of components to enable users pass exams in the learning micro-fromtend application
```
[the remark this answers is not in the document]
Comment (TamoshaytisV) [reply docs/decisions/0001-library-rationale.rst]: fixed

File: docs/decisions/0001-library-rationale.rst [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+(frontend-app-learning).
+
+Main exported component is `SequenceExamWrapper` and it serves as a proxy for the learning app sequence renderer
```
[the remark this answers is not in the document]
Comment (TamoshaytisV) [reply docs/decisions/0001-library-rationale.rst]: fixed

File: docs/decisions/0001-library-rationale.rst [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+with ability to take control over rendering process and show exam related content to users, e.g. exam instructions.
```
[the remark this answers is not in the document]
Comment (TamoshaytisV) [reply docs/decisions/0001-library-rationale.rst]: fixed

**Review by Zacharis278 — APPROVED** [review]
Looks good 🚀

**Pull request merged**